external/urlfrontier/src/test/java/com/digitalpebble/stormcrawler/urlfrontier/StatusUpdaterBoltTest.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.digitalpebble.stormcrawler.urlfrontier;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.digitalpebble.stormcrawler.Metadata;
+import com.digitalpebble.stormcrawler.TestOutputCollector;
+import com.digitalpebble.stormcrawler.TestUtil;
+import com.digitalpebble.stormcrawler.persistence.Status;
+import java.util.HashMap;
+import java.util.concurrent.*;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.config.Configurator;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.tuple.Tuple;
+import org.junit.*;
+import org.junit.rules.Timeout;
+
+public class StatusUpdaterBoltTest {
+    private StatusUpdaterBolt bolt;
+    private TestOutputCollector output;
+    private URLFrontierContainer urlFrontierContainer;
+
+    private static final String persistedKey = "somePersistedKey";
+    private static final String notPersistedKey = "someNotPersistedKey";
+
+    private static ExecutorService executorService;
+
+    @Rule public Timeout globalTimeout = Timeout.seconds(60);
+
+    @BeforeClass
+    public static void beforeClass() {
+        executorService = Executors.newFixedThreadPool(2);
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        executorService.shutdown();
+        executorService = null;
+    }
+
+    @Before
+    public void before() {
+
+        String version = System.getProperty("urlfrontier-version");
+        if (version == null) version = "2.2";
+
+        urlFrontierContainer = new URLFrontierContainer("crawlercommons/url-frontier:" + version);
+        urlFrontierContainer.start();
+
+        bolt = new StatusUpdaterBolt();
+
+        var connection = urlFrontierContainer.getFrontierConnection();
```

Comment:
Why not, the type of the local variable is correctly inferenced and only used locally. 
So there should be no problem.

Reply:
If this really wasn't a proper type I wouldn't have used it. (I hate that too, I even type my Python scripts because of that.)
But yes, that's a brand new feature of Java10. 

I didn't know it either, but it is really convenient for writing code because you don't have to look at the implementation of the function you call. :D